tests/smartclone_test.go

```diff
@@ -0,0 +1,139 @@
+package tests
+
+import (
+	"fmt"
+	"strings"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"kubevirt.io/containerized-data-importer/pkg/controller"
+	"kubevirt.io/containerized-data-importer/tests/framework"
+	"kubevirt.io/containerized-data-importer/tests/utils"
+
+	cdiv1 "kubevirt.io/containerized-data-importer/pkg/apis/core/v1alpha1"
+)
+
+var _ = Describe("[vendor:cnv-qe@redhat.com][level:component]SmartClone tests", func() {
+
+	var sourcePvc *v1.PersistentVolumeClaim
+
+	fillData := "123456789012345678901234567890123456789012345678901234567890"
+	testFile := utils.DefaultPvcMountPath + "/source.txt"
+	fillCommand := "echo \"" + fillData + "\" >> " + testFile
+
+	f := framework.NewFrameworkOrDie("dv-func-test")
+
+	AfterEach(func() {
+		if sourcePvc != nil {
+			By("[AfterEach] Clean up source PVC")
+			err := f.DeletePVC(sourcePvc)
+			Expect(err).ToNot(HaveOccurred())
+			sourcePvc = nil
+		}
+	})
+
+	Describe("Verify DataVolume Smart Cloning - Positive flow", func() {
+		It("succeed creating smart-clone dv", func() {
+			if controller.IsOpenshift(f.K8sClient) {
```

Comment:
So I know I told you guys to use IsOpenshift to skip the test, but I think we can make it more robust by checking if the snapshot storage class we passed in as a parameter exists. That way if the tests are run on a k8s cluster that doesn't have a storage class with csi snapshotting enabled, this will get skipped, and if at some point openshift gets the support needed, we won't have to change this code.

The check can be as simple as taking the class name passed in, and doing a get on the storage class

```go
func (f *Framework) IsSnapshotStorageClassAvailable() bool {
  sc, err := f.K8sClient.StorageV1().StorageClasses().Get(f.snapshotSCName, metav1.GetOptions{})
  if err != nil {
    return false
  }
  return sc.Name == f.snapshotSCName
}